src/PortingAssistantExtensionServer/Services/AnalysisService.cs

```diff
@@ -61,16 +61,16 @@ public async Task<SolutionAnalysisResult> AssessSolutionAsync(AnalyzeSolutionReq
                 var solutionAnalysisResult = await _client.AnalyzeSolutionAsync(request.solutionFilePath, request.settings);
 
                 Collector.SolutionAssessmentCollect(
-                solutionAnalysisResult,
-                runId,
-                triggerType,
-                _request.settings.TargetFramework,
-                PALanguageServerConfiguration.ExtensionVersion,
-                PALanguageServerConfiguration.VisualStudioVersion,
-                DateTime.Now.Subtract(startTime).TotalMilliseconds,
-                PALanguageServerConfiguration.VisualStudioFullVersion,
-                PALanguageServerConfiguration.EnabledDefaultCredentials);
-                CreateClientConnectionAsync(request.PipeName);
+                    solutionAnalysisResult,
+                    runId,
+                    triggerType,
+                    _request.settings.TargetFramework,
+                    PALanguageServerConfiguration.ExtensionVersion,
+                    PALanguageServerConfiguration.VisualStudioVersion,
+                    DateTime.Now.Subtract(startTime).TotalMilliseconds,
+                    PALanguageServerConfiguration.VisualStudioFullVersion,
+                    PALanguageServerConfiguration.EnabledDefaultCredentials);
+
                 return solutionAnalysisResult;
             }
             catch (Exception ex)
```

Comment:
double check front end if we await the assessment response, if we await the rpc channel is better to handle the empty solution result case (pop failed assessment error).

Reply:
With try-catch-finally, the finally() is executed before return. 
For non-exceptions, there is no behavior change from current.

Reply:
As Samm mentioned, I didn't change the way to use rpc channel, but only make sure it's being executed either if we had an exception or finished assessment properly. For this PR's purpose, the focus is to make sure client get notification to unblock our extension client. I agree the concern of what should happen if we failed to assess properly, but I would leave that as a follow up discussion, and probably not an easy solution just by looking at the code here.